internal/app/machined/pkg/system/log/log.go

```diff
@@ -27,14 +27,15 @@ type Log struct {
 
 // New initializes and registers a log for a service.
 func New(name, rootPath string) (*Log, error) {
+	logpath := FormatLogPath(name, rootPath)
+
 	mu.Lock()
-	if l, ok := instance[name]; ok {
+	if l, ok := instance[logpath]; ok {
```

Comment:
the problem here was that log objects for the same service `name` but different `rootPath` were cached as the same object leading to surprising results in the tests when `logPath` is just `tmpDir`